[infra] Enable trimming on Helix for Apple mobile

kotlarmilos opened the pull request:
## Description

This PR (fixes https://github.com/dotnet/runtime/issues/91923) introduces the trimming of library tests on Helix for Apple mobile targets. With trimming enabled, the configuration more closely mirrors the experience of an end user when building Apple mobile applications.

This change results in a `20min (42%) reduction in CI job builds for iOS` (a subset of tests) and a `22min (18%) reduction for tvOS` (the complete coverage).

Initial work is available at https://github.com/dotnet/runtime/pull/92719. 

Test job run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=622954&view=results

kotlarmilos committed:
Add ILLink support to build machines

This commit introduces the EnableAggressiveTrimming parameter in the Apple mobile targets for build machines. It ensures that parameters and items used by the ILLink target are passed correctly.

kotlarmilos committed:
Add ILLink support on Helix

This commit introduces the EnableAggressiveTrimming parameter in the proxy project. Additionally, it ensures that ILLink props and targets are included into the payload.

kotlarmilos committed:
Fix trimming-related test failures

This commit adds ILLink descriptor files to preserve types needed for tests. Additionally, it disables tests that are not trim-compatible.

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos committed:
Fix ILLinkTasksAssembly path in AppleBuild.LocalBuild.props

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos committed:
Override ILLinkTasksAssembly property

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos committed:
Set ILLinkTasksAssembly for NativeAOT tests

kotlarmilos commented:
/azp run runtime-ioslike

mdh1418 commented on src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs:
```diff
@@ -2147,7 +2147,7 @@ private void ValidateGeolocation(IGeolocation location)
         [Fact]
 #if !BUILDING_SOURCE_GENERATOR_TESTS
-        [ActiveIssue("Investigate Build browser-wasm linux Release LibraryTests_EAT CI failure for reflection impl", TestPlatforms.Browser)]
+        [ActiveIssue("Investigate Build browser-wasm linux Release LibraryTests_EAT CI failure for reflection impl", TestPlatforms.Browser | TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.MacCatalyst)]
```
Should we be using `IsBuiltWithAggressiveTrimming` here as well?

kotlarmilos replied:
Thanks, updated.

mdh1418 commented on src/libraries/sendtohelix-mobile.targets:
```diff
@@ -81,7 +81,7 @@
       <HelixCorrelationPayload Include="$(CoreCLRAotSdkDir)"                     Condition="'$(RuntimeFlavor)' == 'coreclr'"
                                                                                  Destination="build/aotsdk" />
-      <HelixCorrelationPayload Include="$(ToolsILLinkDir)"                       Destination="build/microsoft.net.illink.tasks" />
+      <HelixCorrelationPayload Include="$(ToolsILLinkDir)"                       Destination="build" />
```
Is there a particular reason for changing that the ILLink tool lives under `build/microsoft.net.illink.tasks`?

kotlarmilos replied:
The path on Helix wasn't correct and probably not used. The ILLink is copied to the `<HelixCorrelationPayload Include="$(ToolsILLinkDir)" Destination="build" />`.

mdh1418 commented on src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj:
```diff
@@ -49,14 +50,14 @@
   <Target Name="BundleTestAppleApp" DependsOnTargets="AppleBuild" />
 
-  <Target Name="_PrepareForAppleBuildAppOnHelix" DependsOnTargets="_PublishRuntimePack">
+  <Target Name="_PrepareForAppleBuildAppOnHelix">
```
Revisiting this, we moved _PublishRuntimePack to `AppleBuildDependsOn` to make the order of targets more clear. 

Just wondering if this target `_PrepareForAppleBuildAppOnHelix` directly depends on `_PublishRuntimePack` (if so I think we should keep the DependsOnTargets here in addition to adding it to AppleBuildDependsOn).

Are there other targets or parts of the build that depends on `_PublishRuntimePack`? If not, then I think we should keep it clear that we only have `_PublishRuntimePack` for `_PrepareForAppleBuildAppOnHelix` and remove it from `AppleBuildDependsOn`

steveisok replied:
I think it's preference. I like the `AppleBuildDependsOn` because it's one place to see all the targets that run.

mdh1418 replied:
That makes sense, I was thinking that if we only needed `_PublishRuntimePack` for whenever `_PrepareForAppleBuildAppOnHelix` is ran, then we ensure that `_PublishRuntimePack` will only be ran in tandem with `_PrepareForAppleBuildAppOnHelix` (maybe same condition), otherwise over time it might be difficult to simplify. 

e.g. if at some point `_PrepareForAppleBuildAppOnHelix` is no longer needed and is removed, but `_PublishRuntimePack` is kept because it wasn't clear that `_PublishRuntimePack` target was only included to ensure `_PrepareForAppleBuildAppOnHelix` would work properly

mdh1418 commented on src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj:
```diff
@@ -89,6 +90,13 @@
+       <TrimmerRootDescriptor Include="$(ExtraFilesPath)**\*.xml" />
+
+       <!-- To recreate the state of feature flags (RuntimeHostConfigurationOption) during app build-time we need to discard the default setting, coming from SDK, and used the values passed in ProxyProjectForAOTOnHelix.props -->
+       <RuntimeHostConfigurationOption Remove="@(_AppleUsedRuntimeHostConfigurationOption)" />
+       <RuntimeHostConfigurationOption Include="@(_AppleUsedRuntimeHostConfigurationOption)" />
```
Should this be
```suggestion
       <RuntimeHostConfigurationOption Remove="@(RuntimeHostConfigurationOption)" />
       <RuntimeHostConfigurationOption Include="@(_AppleUsedRuntimeHostConfigurationOption)" />
```

ivanpovazan replied:
I will double check, but the idea is to only overwrite the configuration passed on from build machines, while keeping what has been configured for build on helix.

mdh1418 replied:
Maybe I'm missing something, but the original change to me looks like it functionally does nothing? A - B + B?

mdh1418 replied:
Ahh that makes more sense to me. Thank you for adding the example! I totally missed that it would be helpful for updating the metadata values

ivanpovazan replied:
I double-checked this and it is required since some of the feature switches can be specified on the test project level and they will be sent to Helix in form of `_AppleUsedRuntimeHostConfigurationOption`. On the other hand, we cannot *just* use `_AppleUsedRuntimeHostConfigurationOption` instead of `RuntimeHostConfigurationOption` as it does not include the default configuration coming from the SDK.

mdh1418 commented:
Are we going to still run iOSlike tests without `EnableAggressiveTrimming=true`? I think some of the test disabling for iOS may mislead to thinking that the tests don't work on iOS when actually the tests don't work because of trimming. So we should make it clear that those tests are being skipped when trimming.

steveisok commented on eng/testing/tests.mobile.targets:
```diff
@@ -24,7 +24,7 @@
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(EnableAggressiveTrimming)' == 'true' or '$(EnableSoftTrimming)' == 'true'">
+  <PropertyGroup Condition="('$(EnableAggressiveTrimming)' == 'true' or '$(EnableSoftTrimming)' == 'true') and '$(BuildTestsOnHelix)' != 'true'">
```
Because this is in the context of the build machine and not the one on helix. Said another way, we want to support trimming on the build machine for configurations that can handle it.

mdh1418 replied:
I think it would be good to have that as a comment above the property group, to provide more context

steveisok commented:
> Are we going to still run iOSlike tests without `EnableAggressiveTrimming=true`? I think some of the test disabling for iOS may mislead to thinking that the tests don't work on iOS when actually the tests don't work because of trimming. So we should make it clear that those tests are being skipped when trimming.

Since customers could theoretically turn it off, there is value in running w/o trimming. I don't think it needs to be run anywhere except the rolling build.

ivanpovazan commented on eng/testing/tests.ioslike.targets:
```diff
@@ -104,41 +106,69 @@
+      <_ApplePropertyNames Include="HybridGlobalization" />
+      <_ApplePropertyNames Include="InvariantGlobalization" />
+
+      <!-- ILLink properties -->
+      <_ApplePropertyNames Include="_NetCoreAppToolCurrent" />
+      <_ApplePropertyNames Include="DebuggerSupport" />
+      <_ApplePropertyNames Include="EventSourceSupport" />
+      <_ApplePropertyNames Include="JsonSerializerIsReflectionEnabledByDefault" />
+      <_ApplePropertyNames Include="MetadataUpdaterSupport" />
```
I will double check this, these feature flags should not be needed as we are already passing the whole `RuntimeHostConfigurationOption` to helix.

kotlarmilos replied:
Let's try to remove them.

ivanpovazan commented on src/libraries/System.Reflection.MetadataLoadContext/tests/System.Reflection.MetadataLoadContext.Tests.csproj:
```diff
@@ -86,7 +86,7 @@
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetOS)' == 'browser'">
+  <ItemGroup Condition="'$(TargetOS)' == 'browser' or ('$(EnableAggressiveTrimming)' == 'true' and '$(UseNativeAotRuntime)' != 'true')">
```
nit:

We should split this into two ItemGroups:
- one setting up `WasmFilesToIncludeFromPublishDir` only when `Condition="'$(TargetOS)' == 'browser'"`
- and another one which sets `TrimmerRootDescriptor` when `Condition="'$(TargetOS)' == 'browser' or ('$(EnableAggressiveTrimming)' == 'true' and '$(UseNativeAotRuntime)' != 'true')"`

otherwise we would set `WasmFilesToIncludeFromPublishDir` even if it is not used.

ivanpovazan replied:
Fixed

ivanpovazan commented on src/libraries/System.Resources.Extensions/tests/System.Resources.Extensions.Tests.csproj:
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-unix;$(NetFrameworkMinimum)</TargetFrameworks>
     <IncludeRemoteExecutor>true</IncludeRemoteExecutor>
+    <CustomResourceTypesSupport Condition="'$(EnableAggressiveTrimming)' == 'true' and '$(UseNativeAotRuntime)' != 'true'">true</CustomResourceTypesSupport>
```
Additional question:
Does this affect wasm EAT runs as well?

kotlarmilos replied:
Let's check if it is actually needed.

kotlarmilos replied:
This ILLink switch indicates that the trimming of ResourceManager code paths should be avoided.

```	[FAIL] System.Resources.Extensions.Tests.PreserializedResourceWriterTests.ResourceManagerLoadsCorrectReader   Test name: System.Resources.Extensions.Tests.PreserializedResourceWriterTests.ResourceManagerLoadsCorrectReader
[09:30:24.9805730] 2024-04-09 06:30:24.967 System.Resources.Extensions.Tests[60008:82275223]    Assembly:  [System.Resources.Extensions.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51]
[09:30:24.9805830] 2024-04-09 06:30:24.967 System.Resources.Extensions.Tests[60008:82275223]    Exception messages: System.NotSupportedException : Use of ResourceManager for custom types is disabled. Set the MSBuild Property CustomResourceTypesSupport to true in order to enable it.

ivanpovazan commented on src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj:
```diff
@@ -68,7 +68,7 @@
     <ProjectReference Include="ApplyUpdate\System.Reflection.Metadata.ApplyUpdate.Test.NewMethodThrows\System.Reflection.Metadata.ApplyUpdate.Test.NewMethodThrows.csproj" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetOS)' == 'browser'">
+  <ItemGroup Condition="'$(TargetOS)' == 'browser' or ('$(EnableAggressiveTrimming)' == 'true' and '$(UseNativeAotRuntime)' != 'true')">
```
I made a similar comment above.

kotlarmilos replied:
Thanks, updated.

ivanpovazan commented on src/libraries/System.Runtime/tests/System.Runtime.Tests/ILLink.Descriptors.iOS.xml:
```diff
@@ -0,0 +1,40 @@
+    <type fullname="System.Tests.EnumTests" />
+    <type fullname="System.Tests.ArrayTests" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Nullable`1[[System.Int32,System.Private.CoreLib]],System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Byte, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.SByte, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.UInt16, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Int16, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.UInt32, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Int32, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.UInt64, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Int64, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.IntPtr, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.UIntPtr, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Char, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Boolean, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Single, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Double, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Decimal, System.Private.CoreLib]]" />
+    <type fullname="System.Reflection.Tests.InvokeRefReturnNetcoreTests+TestClass`1[[System.Reflection.BindingFlags, System.Private.CoreLib]]" />
```
I don't think we need this and we should just root the generic type (not the instantiations).

ivanpovazan commented on src/libraries/System.Runtime/tests/System.Runtime.Tests/ILLink.Descriptors.iOS.xml:
```diff
@@ -0,0 +1,40 @@
+    <type fullname="System.Reflection.Tests.MethodBaseTests" />
+    <type fullname="System.Tests.Types.ModifiedTypeTests+ModifiedTypeHolder" />
+    <type fullname="System.Tests.ArrayTests" />
+    <type fullname="System.Tests.ArrayTests+GenericStruct`1[[System.Int32,System.Private.CoreLib]][]" />
```
Same as above.

ivanpovazan commented on src/mono/msbuild/apple/build/AppleBuild.LocalBuild.targets:
```diff
@@ -34,8 +34,59 @@
+    <NoWarn>$(NoWarn);IL2103;IL2105;IL2025;IL2111</NoWarn>
+    <!-- Warns about missing Xamarin.iOS.dll -->
+    <NoWarn Condition="'$(EnableSoftTrimming)' == 'true'">$(NoWarn);IL2035</NoWarn>
+    <!-- https://github.com/dotnet/sdk/issues/18581 tracks needing to set 2 properties to disable warnings. -->
+    <EnableTrimAnalyzer>false</EnableTrimAnalyzer>
```
This seems redundant as the issue has been resolved.

ivanpovazan commented on src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj:
```diff
@@ -99,10 +107,22 @@
 
+  <Target Name="_TrimApp" DependsOnTargets="ILLink;CopyFilesToPublishDirectory">
+    <ItemGroup>
+      <AppleAssembliesToBundle Remove="@(AppleAssembliesToBundle)" />
```
As we are only adjusting the `AppleAssembliesToBundle` item group, I don't think it is necessary to have a dedicated build target for that. 
Although, it is possible there isn't a better place to do so. 
In any case, the name does not seem suitable (something like `_AdjustTrimmedAssembliesToBundle` might be a better option, but I don't have a strong opinion).

ivanpovazan replied:
Some additional thoughts..

We reached a point where we managed to enable trimming on Helix and things work, which is great as it was much needed and long awaited feature. However, some parts of this implementation could be improved in general. 

One of the core problems was splitting up the build/publish across build machines and Helix.  That involved:
- recreating the build folder structure on Helix, but also 
- configuring the build project "to resume" app trimming and deployment.

The problem that remains - which we did manage to workaround - is that we are reusing the `publish` folder for different things and rewriting it during the build on Helix. That is not ideal, will be hard to maintain and should be improved in the future. 
Shuffling `AppleAssembliesToBundle` here is done for this reason, we have an initial set of assemblies to bundle, but during trimming some might be removed, so we have to manually adjust this item group. 

Another issue with reusing the publish folder for everything is that we place the `ProxyProjectForAOTOnHelix.props`, `ProxyProjectForAOTOnHelix.proj` and many other files in it, so we can end up bundling them into the final app bundle, which is again unnecessary and error prone. It is true we have some file filtering in `AppleAppBuilder` task to prevent bundling unnecessary files, but that shouldn't be the place to fix these kind of problems. 

I propose we handle these improvements as a follow-up task (and not delay this PR further) but we should make sure we have a tracking issue for it. @kotlarmilos what do you think?

PS For now, I can also add a comment in the code to explain the reason for `_AdjustTrimmedAssembliesToBundle`

ivanpovazan replied:
Add a task in the linked issue and a comment describing what is the purpose of `_AdjustTrimmedAssembliesToBundle`

ivanpovazan reviewed (commented):
@kotlarmilos thanks for ironing out the PR!
I left a few comments and will double check what Mitch was asking about `RuntimeHostConfigurationOptions` so we can merge it soon.

kotlarmilos commented on src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj:
```diff
@@ -99,10 +107,22 @@
     </ItemGroup>
 
+    <ItemGroup>
+      <ReferencePath Include="@(AppleReferenceSharedPathFiles->'$(MicrosoftNetCoreAppRuntimePackLibDir)%(FileName)%(Extension)')" />
+      <ReferencePath Include="@(AppleReferenceExtraPathFiles->'$(ExtraFilesPath)%(FileName)%(Extension)')" />
```
Yes, you are right. It is used in the _RunILLink target. I can moved it outside of the target, as it is not related to portable runtime pack target.

kotlarmilos committed:
Remove ILLink properties as they may not be necessary

kotlarmilos committed:
Remove redundant properties and add comments

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos commented:
Thank you for the feedback. I've removed redundant ILLink properties and addressed the infrastructure comments. To ensure that tests can run without trimming on rolling builds, I suggest updating disabled tests to be conditioned by `IsBuiltWithAggressiveTrimming` which can utilize `IsEnvironmentVariableTrue`.

ivanpovazan commented on src/mono/msbuild/apple/build/AppleBuild.LocalBuild.targets:
```diff
@@ -34,8 +34,59 @@
+           TrimMode="" is needed so the root assemblies are correctly identified -->
+      <AppleAssembliesToBundle TrimMode="" Condition="'%(FileName)' == '$(AssemblyName)'" />
+
+      <!-- TODO: find out why these assemblies aren't copied by copyused even though they're referenced -->
```
It seems like we don't. I am not 100% sure but it seems like `EnableSoftTrimming` was a result of previous attempts to make trimming work, and with https://github.com/dotnet/runtime/commit/8c6aeea1d8e17fd692df0f509d415d9ad79d83bc seems it got disabled even on CI. 
At the same time, searching through the current state of the repository we don't seem to specify it anywhere.

@steveisok should we just remove it altogether?

steveisok replied:
Yeah, feel free to remove it. We can always bring it back if need be.

kotlarmilos committed:
Revert ILLink switch properties

kotlarmilos commented:
/azp run runtime-ioslike

ivanpovazan committed:
Split browser-specific configuration

ivanpovazan committed:
Add description about _AdjustTrimmedAssembliesToBundle

ivanpovazan commented:
/azp run runtime-ioslike

kotlarmilos committed:
Use ActiveIssue with IsBuiltWithAggressiveTrimming for disabled tests

kotlarmilos committed:
Trim tests only on PR builds

kotlarmilos committed:
Enable aggressive trimming by default

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos commented on src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs:
```diff
@@ -2118,7 +2118,7 @@ public void ComplexObj_As_Enumerable_Element()
 
 #if !BUILDING_SOURCE_GENERATOR_TESTS
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser), nameof(PlatformDetection.IsNotAppleMobile))]
```
Thanks, updated.

kotlarmilos commented on src/libraries/System.Reflection.Metadata/tests/Metadata/Decoding/SignatureDecoderTests.cs:
```diff
@@ -129,7 +129,7 @@ public static void VarArgsCaller()
 
         // Test as much as we can with simple C# examples inline below.
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.HasAssemblyFiles))]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.HasAssemblyFiles), nameof(PlatformDetection.IsNotAppleMobile))]
```
Thanks, updated.

kotlarmilos commented on src/libraries/System.Reflection.TypeExtensions/tests/ModuleTests.cs:
```diff
@@ -24,6 +24,7 @@ public void GetModuleVersionId_HasModuleVersionId_BehaveConsistently()
 
         // This calls Assembly.Load, but xUnit turn is into a LoadFrom because TinyAssembly is just a Content item in the project.
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsAssemblyLoadingSupported))]
+        [SkipOnPlatform(TestPlatforms.Browser | TestPlatforms.iOS | TestPlatforms.MacCatalyst | TestPlatforms.tvOS, "TinyAssembly GUID differs when trimming is enabled.")]
```
I believe it is possible. We can add something like:
```csharp
if (!(PlatformDetection.IsMonoRuntime && PlatformDetection.IsAppleMobile && PlatformDetection.IsBuiltWithAggressiveTrimming))
{
    Assert.Equal(Guid.Parse("{06BB2468-908C-48CF-ADE9-DB6DE4614004}"), module.GetModuleVersionId());
}
```

kotlarmilos commented on src/libraries/System.Runtime/tests/System.Runtime.Tests/System/NullableMetadataTests.cs:
```diff
@@ -132,6 +132,7 @@ public static void NullableAttributesOnPublicApiOnly(Type type)
         [Theory]
         [InlineData("mscorlib")]
         [InlineData("System.Threading.Overlapped")]
+        [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.MacCatalyst, "ILLink doesn't root forwarded types from rooted assemblies: https://github.com/dotnet/runtime/issues/99592")]
```
Thanks, updated.

kotlarmilos committed:
Fix ident

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos committed:
Fix buildArgs command

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos committed:
Fix ident

kotlarmilos committed:
Fix ident

kotlarmilos commented:
/azp run runtime-ioslike

kotlarmilos commented:
> Since customers could theoretically turn it off, there is value in running w/o trimming. I don't think it needs to be run anywhere except the rolling build.

The trimming is enabled by default on PR builds and disabled on rolling builds.

ivanpovazan reviewed (approved):
I believe the only thing that is left is to remove references to `EnableSoftTrimming`.
Other than that I think we are good to go!

kotlarmilos committed:
Remove EnableSoftTrimming property

kotlarmilos commented:
/azp run runtime-extra-platforms

steveisok reviewed (approved):
Nice job!
